### PR TITLE
Update dependencies and make necessary code adjustments

### PR DESCRIPTION
**Pull request by dcoutts**
Update to the latest versions of ledger, network, consensus etc.

Some code changes required.

### CODE REVIEW

File: cardano-config/src/Cardano/Config/Shelley/Orphans.hs [hunk 1]
```diff
@@ -81,7 +85,7 @@ instance Crypto crypto => FromJSON (ShelleyGenesis crypto) where
         <*> obj .: "MaxHeaderSize"
         <*> obj .: "GenDelegs"
         <*> obj .: "InitialFunds"
-
+        <*> pure emptyGenesisStaking  --TODO
```
Comment (dcoutts) [inline cardano-config/src/Cardano/Config/Shelley/Orphans.hs:88]: This turns out to  be quite big, and ends up needing extra conversion functions in the ledger for opaque types. So these changes cannot be included in this PR.

File: cardano-config/test/Test/Cardano/Config/Gen.hs [hunk 2]
```diff
@@ -74,6 +80,7 @@ genShelleyGenesis =
     <*> fmap fromIntegral (Gen.word $ Range.linear 100 1000000)
     <*> fmap Map.fromList genGenesisDelegationList
     <*> fmap Map.fromList genFundsList
+    <*> pure emptyGenesisStaking
```
Comment (dcoutts) [inline cardano-config/test/Test/Cardano/Config/Gen.hs:83]: Same here. I have generators in another branch, but the JSON to/from cannot be completed yet, so these tests would not pass yet.

**Comment by erikd** [comment]
This is failing the `chairman` test and I have tracked it down to:
```
[cardano_byron_proxy.shelley.client:Notice:45] [2001-09-09 02:01:12.70 UTC] IP 127.1.0.5:3005 ErrorPolicySuspendConsumer (Just (ConnectionExceptionTrace Network.Socket.connect: <socket: 25>: does not exist
```
which is in the failed test run bit not in a successful run.

**Comment by erikd** [comment]
The socket failure is a complete red herring. The real problem is:
```
[  140.026314] jmhmrp2nwyxjk31kf35w8pv77dq6hjyl-unit-script-cardano-node_-start[729]: NoDbMarkerAndNotEmpty "/var/lib/cardano-node/db-5/protocolMagicId"
```
which is an error coming out of the `ouroborous-consensus` storage layer. People have been notified.

**Comment by dcoutts** [comment]
bors merge

**Comment by dcoutts** [comment]
bors merge

**Pull request merged**